packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -302,7 +306,7 @@ class S3UploadTask {
       builder
         ..bucket = _bucket
         ..body = body
-        ..contentType = body.contentType
+        ..contentType = body.contentType ?? fallbackContentType
```

Comment:
But if I pass `AWSFile.fromPath('myfile.txt')` wouldn't this get a content type of `application/octet-stream`?

Reply:
I had similar question, from description of "when content type is not determinable" I was expecting to see some automatic determination of type which I don't see, though very possible I'm missing something, or not sure if automatic determination was the intention of flutter plugin caller?

Reply:
> But if I pass AWSFile.fromPath('myfile.txt') wouldn't this get a content type of application/octet-stream?

Yes that's correct.

>from description of "when content type is not determinable" I was expecting to see some automatic determination of type which I don't see, hough very possible I'm missing something, or not sure if automatic determination was the intention of flutter plugin caller?

My description is confusing sorry. Amplify libraries don't provide functionality to auto detect the mime type of a file to be uploaded. If library consumer doesn't specify `contentType` calling upload APIs, the `contentType` will be `application/octet-stream` as arbitrary binary data. Note that this won't change the file content, when proper extension name is provided, the file consumer can still interpret the file content correctly, including use in a file system, or rendering on Web (svg, jpg, png etc.).

Auto mime type detection is tricky to implement as there is not a "standard" to guide the implementation, I'm happy to sync up offline regarding this topic.